modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -9,6 +9,18 @@ import Foundation
 import MobileCoreServices
 #endif
 
+public protocol NetworkSession {
+    func loadData(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask
+}
+
+extension URLSession: NetworkSession {
+    public func loadData(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
+        let task = dataTask(with: request, completionHandler: completionHandler)
+        task.resume()
```

Comment:
@leszek-s I think we shouldn't resume the request here, because this is an unexpected side effect.
We should try to mimic the URLSession API as much as we can.

Reply:
If the methods in the protocol, are exactly the same as the URLSession correspondent one's, we don't need to implement the methods, because the mapping is automatic.

https://christiantietze.de/posts/2016/02/mock-replace-adapter/

Reply:
@4brunu I've put it here to give a possibility to have a better control on when the task is started if someone for example wants to delay some tasks. However i'm still thinking about some other solutions.

Reply:
To do that, maybe we can override the method resume and add the logic there.

Reply:
resume() is a method of URLSessionDataTask and not URLSession so we can't do that. Do you think it would be a better idea to add a method to this NetworkSession/URLSessionProtocol protocol for example resumeTask(dataTask: URLSessionDataTask) with a default implementation and use that instead of calling resume on dataTask?

Reply:
OK I removed this resume() and went with simple URLSessionProtocol with the same method as original. I currently do not have a better idea how to handle that resume method so maybe lets just do it with a minimal change as originally propose in the issue. I think delaying task should still somehow be possible by overriding execute method so for now we can live without that control over resume().